include/zephyr/drivers/stepper_motor_controller.h

```diff
@@ -0,0 +1,257 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_MOTOR_CONTROLLER_H
+#define ZEPHYR_INCLUDE_STEPPER_MOTOR_CONTROLLER_H
+
+/** \addtogroup stepper_motor_controller
+ *  @{
+ */
+
+/**
+ * @file
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Stepper Motor Position Format required while setting or getting stepper motor positions.
+ */
+enum micro_step_resolution {
+	FULL_STEP,
+	MICRO_STEP_2,
+	MICRO_STEP_4,
+	MICRO_STEP_8,
+	MICRO_STEP_16,
+	MICRO_STEP_32,
+	MICRO_STEP_64,
+	MICRO_STEP_128,
+	MICRO_STEP_256,
+};
+
+/**
+ * @typedef stepper_motor_controller_set_steps_to_move_t
+ * @brief Stepper Motor Controller shall move the motor with given steps
+ */
+typedef int32_t (*stepper_motor_controller_set_steps_to_move_t)(const struct device *dev,
+								int32_t steps);
+
+/**
+ * @typedef stepper_motor_controller_set_velocity_t
+ * @brief Set the velocity in steps per seconds. For controllers such as DRV8825 where you
+ * toggle the STEP Pin, the pulse_length would have to be calculated based on this parameter in the
+ * driver. For controllers where velocity can be set, this parameter corresponds to max_velocity
+ */
+typedef int32_t (*stepper_motor_controller_set_velocity_t)(const struct device *dev,
+							   uint32_t steps_per_sec);
```

Comment:
```suggestion
typedef int32_t (*stepper_motor_controller_set_velocity_t)(const struct device *dev,
							   uint32_t steps_per_nanosecond);
```
Not sure if limitting here to steps per second is applicable to most applications. Because float would not be an option here fractions of steps or seconds are not possible.

The Nanoseconds i would have assumed to be a o.k. option because for example pwm.h .
Additional macros for conversion could be provided too, in reference to K_MSEC or K_SECONDS.

Reply:
`steps_per_nanosecond`would be a bit too much when I would like to move, say 1 step/sec. But now that I think, how would I move half a step a second?

Reply:
Seems like the speed should be set in terms of clock used by the controller. In the case of the TMC, its usually defined as microsteps/time-unit.